### PR TITLE
Mac support

### PR DESCRIPTION
**Pull request by fracai**
Mac support by using the `say` command.
Also adds '--' to relevant tts commands to protect text beginning with a dash.

### CODE REVIEW

File: ipod-shuffle-4g.py [hunk 1]
```diff
@@ -97,12 +97,19 @@ def group_tracks_by_id3_template(tracks, template):
...
 
 class Text2Speech(object):
-    valid_tts = {'pico2wave': True, 'RHVoice': True, 'espeak': True}
+    valid_tts = {'pico2wave': True, 'RHVoice': True, 'espeak': True, 'say': True}
 
     @staticmethod
     def check_support():
         voiceoverAvailable = False
 
+        # Check for say voiceover
```
Comment (NicoHood) [inline ipod-shuffle-4g.py:106]: I recommend adding a note, that say is a mac specific tool

File: ipod-shuffle-4g.py [hunk 2]
```diff
@@ -464,7 +480,7 @@ def __init__(self, parent):
     def set_master(self, tracks):
         # By default use "All Songs" builtin voiceover (dbid all zero)
         # Else generate alternative "All Songs" to fit the speaker voice of other playlists
-        if self.playlist_voiceover and (Text2Speech.valid_tts['pico2wave'] or Text2Speech.valid_tts['espeak']):
+        if self.playlist_voiceover and (Text2Speech.valid_tts['pico2wave'] or Text2Speech.valid_tts['espeak'] or Text2Speech.valid_tts['say']):
```
Comment (NicoHood) [inline ipod-shuffle-4g.py:483]: I am wondering if this additional and check is even required at all. But looking at the source it turns out there is lots of other stuff that can be improved, unrelated to your changes. So just treat this comment as me thinking out loud.

**Comment by NicoHood** [comment]
I think we can safely merge this @nims11. I also like the `--` improvement, good idea!

File: ipod-shuffle-4g.py (hunk 2 above)
Comment (fracai) [reply ipod-shuffle-4g.py:483]: There's another fork out there that improves TTS that I think would be nice to bring in.

Comment (NicoHood) [reply ipod-shuffle-4g.py:483]: Can you point me to that fork?

Comment (fracai) [reply ipod-shuffle-4g.py:483]: https://github.com/thomas-hori/IPod-Shuffle-4g/commit/0a72c1102d25101749279f6071ebacdd7f1ca7eb

File: ipod-shuffle-4g.py (hunk 1 above)
Comment (fracai) [reply ipod-shuffle-4g.py]: I added to the comment and warning message.

Comment (NicoHood) [reply ipod-shuffle-4g.py]: Thanks! I think it also makes sense to add it to the readme.

Comment (fracai) [reply ipod-shuffle-4g.py]: I added a brief mention of "say" support to the readme. I couldn't find an up to date man page or relevant reference online and aside from mentioning that it's available, I can't think of what else to mention. Suggestions welcome :-)

**Comment by NicoHood** [comment]
Thanks! I would wait for @nims11 feedback and then merge it.

**Review by NicoHood — APPROVED** [review]

**Review by nims11 — APPROVED** [review]
LGTM, seems to be working on my macos

**Pull request merged**